packages/annotations/src/vote.ts

```diff
@@ -0,0 +1,97 @@
+/* Copyright (c) 2018 Environmental Systems Research Institute, Inc.
+ * Apache-2.0 */
+
+import {
+  ArcGISRequestError,
+  ArcGISAuthError,
+  IRequestOptions
+} from "@esri/arcgis-rest-request";
+
+import {
+  addFeatures,
+  IAddFeaturesRequestOptions,
+  IAddFeaturesResult
+} from "@esri/arcgis-rest-feature-service";
+
+import { IAnnoFeature } from "./add";
+import { UserSession } from "@esri/arcgis-rest-auth";
+import { searchAnnotations } from "./search";
+
+export interface IVoteRequestOptions extends IRequestOptions {
+  url: string;
+  annotation: IAnnoFeature;
+  /**
+   * Set to 'true' in order to vote another annotation down.
+   */
+  downVote?: boolean;
+  authentication: UserSession;
+}
+
+/**
+ * ```js
+ * import { vote } from "@esri/hub-annotations";
+ * //
+ * voteOnAnnotation({
+ *   url,
+ *   annotation: { attributes: {} },
+ *   authentication
+ * })
+ *   .then(response);
+ * ```
+ * Upvote (or downvote) another annotation.
+ * @param requestOptions - request options that may include authentication
+ * @returns A Promise that will resolve with the response from the service after attempting to vote on an annotation.
+ */
+export function voteOnAnnotation(
+  requestOptions: IVoteRequestOptions
+): Promise<IAddFeaturesResult> {
+  const annotation = requestOptions.annotation;
+  const auth = requestOptions.authentication as UserSession;
+  const url = requestOptions.url;
+
+  if (!auth) {
+    return new Promise((resolve, reject) => {
+      reject(
+        new ArcGISAuthError(`Voting by anonymous users is not supported.`)
+      );
+    });
+  }
+
+  if (annotation.attributes.author === auth.username) {
+    return new Promise((resolve, reject) => {
+      reject(
+        new ArcGISRequestError(`Users may not vote on their own comment/idea.`)
+      );
+    });
+  }
+
+  return searchAnnotations({
+    url,
+    where: `parent_id = ${annotation.attributes.OBJECTID} AND author = '${
+      auth.username
+    }'`
+  }).then(searchResponse => {
+    // dont let folks vote on an idea more than once
```

Comment:
What if they want to change their vote from an upvote to a downvote? We don't (yet) provide an API to remove the previous vote first, so it seems that unless you know the implementation details you can't really change your vote. Is that intentional?